### PR TITLE
Newlintmodular

### PR DESCRIPTION
**Pull request by tnarg999**
This PR replaces the `lintmodular` branch PR since the merge conflicts on that branch were too hard to resolve

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -0,0 +1,9 @@
+github.com/grantae/certinfo v0.0.0-20170412194111-59d56a35515b h1:NGgE5ELokSf2tZ/bydyDUKrvd/jP8lrAoPNeBuMOTOk=
+github.com/grantae/certinfo v0.0.0-20170412194111-59d56a35515b/go.mod h1:zT/uzhdQGTqlwTq7Lpbj3JoJQWfPfIJ1tE0OidAmih8=
+golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2/go.mod h1:djNgcEr1/C05ACkg1iLfiJU5Ep61QUkGW8qpdssI0+w=
+golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a h1:vclmkQCjlDX5OydZ9wv8rBCcS0QyQY66Mpf/7BZbInM=
+golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a/go.mod h1:LzIPMQfyMNhhGPhUkYOs5KpL4U8rLKemX1yGLhDgUto=
+golang.org/x/net v0.0.0-20190404232315-eb5bcb51f2a3/go.mod h1:t9HGtf8HONx5eT2rtn7q6eTqICYqUVnKs3thJo3Qplg=
+golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a/go.mod h1:STP8DvDyc/dI5b8T5hshtkjS+E42TnysNCUPdjciGhY=
+golang.org/x/sys v0.0.0-20190412213103-97732733099d/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
+golang.org/x/text v0.3.0/go.mod h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ=
```
Comment (aokeke) [inline go.sum:9]: Where do all these values come from? And should they be pushed to a public repo. Just making sure it isn't any sensitive information.

File: ocsptools/ocsp_tools.go [hunk 2]
```diff
@@ -1,16 +1,21 @@
 package ocsptools
 
 import (
-	"../linter"
 	"bytes"
 	"crypto"
 	"crypto/x509"
 	"encoding/base64"
 	"fmt"
+	"github.com/googleinterns/ocsp-response-linter/linter"
```
Comment (aokeke) [inline ocsptools/ocsp_tools.go:9]: Same comment here.

File: ocsptools/ocsp_tools.go [hunk 3]
```diff
@@ -111,12 +116,25 @@ func CreateOCSPReqFromCert(certFile string, ocspURL string, reqMethod string, ha
 
 // getOCSPResponse constructs and sends an OCSP request then returns the OCSP response
 func GetOCSPResponse(ocspReq *http.Request) ([]byte, error) {
+	startTime := time.Now()
+
 	httpClient := &http.Client{}
 	httpResp, err := httpClient.Do(ocspReq)
 	if err != nil {
 		return nil, fmt.Errorf("Error sending http request: %w", err)
 	}
```
Comment (aokeke) [inline ocsptools/ocsp_tools.go]: Will this ever timeout? Is there a way to set a timeout or have this automatically fail if this has gone beyond the time limit? It might be complicated, but possibly running in a thread and then killing the thread if it hasn't finished in the time limit. If it is too complicated, it can be saved for another PR.

File: test/lint_test.go [hunk 4]
```diff
@@ -4,7 +4,7 @@ import (
 	"testing"
 	"golang.org/x/crypto/ocsp"
 	"io/ioutil"
-	"../linter"
+	"github.com/googleinterns/ocsp-response-linter/linter"
```
Comment (aokeke) [inline test/lint_test.go:7]: same commend here about using url paths.

**Review by aokeke — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (tnarg999) [reply go.sum:9]: These values are generated from `go mod init`. My only understanding of this file is that it plays a part in managing package dependencies. It should be safe to commit (e.g. https://github.com/zmap/zlint/blob/master/v2/go.sum)

File: ocsptools/ocsp_tools.go (hunk 2 above)
Comment (tnarg999) [reply ocsptools/ocsp_tools.go:9]: I learned from talking with Filippo today that using relative paths is not a good practice and instead I should be using urls like this.

File: ocsptools/ocsp_tools.go (hunk 3 above)
Comment (tnarg999) [reply ocsptools/ocsp_tools.go]: The HTTP Client does allow you to set a timeout, I should probably use that field.

**Review by aokeke — APPROVED** [review]
LGTM, it still seems weird to me to use a url for dependencies within the project because then if you're trying to test changes locally, you have to remember to change those to point to local values.
Also if you don't have a network connection, you can't use the tool. Then again if you don't have a network connection you probably can't use the tool anyway.

I'll defer Filippo's judgement.

**Comment by tnarg999** [comment]
Yeah I definitely had some trouble figuring out how to test locally with this setup, turns out you have to reroute your packages in the `go.mod` files to point to your local versions. I'll check back in with Filippo to ask him why Go has been set up this way.

File: linter/linter.go [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
 
-const (
-	ProducedAtLimit = "96h" // 4 days
-	ThisUpdateLimit = "96h" // 4 days
-)
+var StatusIntMap = map[int]string {
+	ocsp.Good: "good",
+	ocsp.Revoked: "revoked",
+	ocsp.Unknown: "unknown",
+	// ocsp.SeverFailed is never used:
```
Comment (wuhelena) [inline linter/linter.go]: Could you provide more explanation for why ServerFailed is never used?

File: linter/lintfuncs.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	"golang.org/x/crypto/ocsp"
+	"time"
+)
+
+const (
+	ProducedAtLimit = "96h" // 4 days
+	ThisUpdateLimit = "96h" // 4 days
+)
+
+func LintProducedAtDate(resp *ocsp.Response) error {
```
Comment (wuhelena) [inline linter/lintfuncs.go:14]: Please add function header comment

File: linter/lintfuncs.go [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	if err != nil {
+		return err
+	}
+	if time.Since(resp.ProducedAt) > limit {
+		return fmt.Errorf("OCSP Response producedAt date is more than %s in the past", ProducedAtLimit)
+	}
+	return nil
+}
+
+func LintThisUpdateDate(resp *ocsp.Response) error {
```
Comment (wuhelena) [inline linter/lintfuncs.go:25]: Ditto above

File: linter/linter.go (hunk 5 above)
Comment (tnarg999) [reply linter/linter.go]: Will add https://godoc.org/golang.org/x/crypto/ocsp#pkg-constants in comment
